src/common.h

```diff
@@ -31,6 +31,34 @@ extern "C" {
 	"Please do not compile this with a lslboost version older than 1.45 because the library would otherwise not be protocol-compatible with builds using other versions."
 #endif
 
+// size of a cache line
+#if defined(__s390__) || defined(__s390x__)
+#define CACHELINE_BYTES 256
+#elif defined(powerpc) || defined(__powerpc__) || defined(__ppc__)
+#define CACHELINE_BYTES 128
+#else
+#define CACHELINE_BYTES 64
+#endif
+
+// force-inline the given function, if possible
```

Comment:
IIRC Boost has macros for both forceinline and likely/unlikely but with more supported compilers

Reply:
Fair point - will use those. For likely/unlikely, I'd expect us to sprinkle that in along the "hot" path here and there over time. There's a bit of an issue with the boost definitions, which differ from how e.g. the Linux kernel or UE4 define them, in that they don't include the double-not. That causes them to be prone to accidentally wrong usage (e.g., if I were to replace `if (myptr)` by `if (BOOST_LIKELY(myptr))`, that would expand to `if (__builtin_expect(myptr, 1))` -- i.e., it assumes that the pointer is 1, rather than that the expression evaluates to true in a boolean context (causing it to fail to optimize, and in some cases might even pessimize the code path). We could try to add that to our mental checklist when reviewing code that uses that, but it feels more prudent to write it into the macro since otherwise it'd put extra burden on the devs/reviewers. One might even argue that the boost versions are misleading names for something that would be better described as `BOOST_LIKELYONE` and `BOOST_LIKELYZERO`.

But to gain the extra compilers supported by the boost versions while keeping the ease of use we could define it here as
```
#define LIKELY(x) BOOST_LIKELY(!!(x))
#define UNLIKELY(x) BOOST_UNLIKELY(!!(x))
```

For the forceinline, do you prefer using the boost version in the code or should we alias that here similarly to a shorter macro name, as in
```
#define FORCEINLINE BOOST_FORCEINLINE
```

Reply:
I checked the boost source code; BOOST_LIKELY is defined for mostly GCC/Clang and the Intel and IBM compilers. MSVC waits for C++20's `[[likely]]` attribute and wants assumes the condition in `if()` to be true.
So I'm for `#define LIKELY(x) BOOST_LIKELY(!!(x))` and using that everywhere. There's just one `UNLIKELY` that'd need to be changed in the code below. As for `BOOST_FORCEINLINE`, I'd just use that.

Reply:
Great, will do that